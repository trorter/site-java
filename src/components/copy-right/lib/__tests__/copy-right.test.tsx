import React from "react";
import CopyRight from "../copy-right";
import {create} from "react-test-renderer";

describe("Should render", () => {
    describe("Without props", () => {

        it("and pass the snapshot test", () => {
            const view = create(<CopyRight />).toJSON();
            expect(view).toMatchSnapshot();
        })
    })

    describe("With props", () => {

        it("and pass the snapshot test", () => {
            const extended = true;

            const view = create(<CopyRight extended={extended}/>).toJSON();
            expect(view).toMatchSnapshot();
        })
    })
})