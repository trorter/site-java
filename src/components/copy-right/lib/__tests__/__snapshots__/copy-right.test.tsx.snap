// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render With props and pass the snapshot test 1`] = `
<div
  className="ant-space css-dev-only-do-not-override-yp8pcc ant-space-vertical"
  style={Object {}}
>
  <div
    className="ant-space-item"
    style={
      Object {
        "marginBottom": 8,
      }
    }
  >
    <div
      className="ant-typography css-dev-only-do-not-override-yp8pcc"
      style={
        Object {
          "WebkitLineClamp": undefined,
          "color": "inherit",
          "textAlign": "left",
        }
      }
    >
      Copyright © 2023 by Andrei Ledovskikh
    </div>
  </div>
  <div
    className="ant-space-item"
    style={Object {}}
  >
    <div
      className="ant-typography css-dev-only-do-not-override-yp8pcc"
      style={
        Object {
          "WebkitLineClamp": undefined,
          "color": "inherit",
          "textAlign": "justify",
        }
      }
    >
      All rights reserved. No part of this publication may be reproduced, distributed, or transmitted in any form or by any means, including photocopying, recording, or other electronic or mechanical methods, without the prior written permission of the publisher, except in the case of brief quotations embodied in critical reviews and certain other noncommercial uses permitted by copyright law. For permission requests, write to the following email: an.y.ledov@gmail.com.
    </div>
  </div>
</div>
`;

exports[`Should render Without props and pass the snapshot test 1`] = `
<div
  className="ant-space css-dev-only-do-not-override-yp8pcc ant-space-vertical"
  style={Object {}}
>
  <div
    className="ant-space-item"
    style={Object {}}
  >
    <div
      className="ant-typography css-dev-only-do-not-override-yp8pcc"
      style={
        Object {
          "WebkitLineClamp": undefined,
          "color": "inherit",
          "textAlign": "left",
        }
      }
    >
      Copyright © 2023 by Andrei Ledovskikh
    </div>
  </div>
</div>
`;
